trino-s3-proxy/src/main/java/io/trino/s3/proxy/server/rest/TrinoS3ProxyResource.java

```diff
@@ -45,7 +44,7 @@ public TrinoS3ProxyResource(SigningController signingController, TrinoS3ProxyCli
     {
         this.signingController = requireNonNull(signingController, "signingController is null");
         this.proxyClient = requireNonNull(proxyClient, "proxyClient is null");
-        this.serverHostName = requireNonNull(trinoS3ProxyConfig, "restConfig is null").getHostName();
+        this.serverHostName = requireNonNull(trinoS3ProxyConfig, "trinoS3ProxyConfig is null").getS3HostName();
```

Comment:
We don't usually check for null on any object we're using - i.e. config objects. You'd get an early NPE anyway.

Reply:
I was debating this one, thanks for clarifying